src/collection-browser.ts

```diff
@@ -2086,6 +2097,24 @@ export class CollectionBrowser
     }
   }
 
+  /**
+   * Fetches search results for privileged users when in manage view
+   * If total results exceed the threshold, partially resets the datasource pages
+   * and fetches the first page with a limit based on the threshold
+   */
+  private fetchManagableSearchResults(): void {
+    const maxAllowedResults = this.maxPagesToManage * this.pageSize;
+    if (
+      this.pageContext === 'search' &&
+      this.dataSource.totalResults > maxAllowedResults &&
```

Comment:
Potential blocker: I'm not sure about the `this.dataSource.totalResults > maxAllowedResults` condition. That will prevent the larger fetch if there are, say, only 500 total results, right?

I think we can remove that condition, because we still want to fetch all the results in that case (and requesting 750 would still be fine)

Reply:
Make sense. I just need to change this logic a bit to make is more accurate.